### PR TITLE
luaPackages: update

### PR DESCRIPTION
**Pull request by spencerpogo**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/lua-modules/generated-packages.nix [hunk 1]
```diff
@@ -184,7 +184,7 @@ buildLuarocksPackage {
   pname = "cassowary";
   version = "2.3.2-1";
   knownRockspec = (fetchurl {
-    url    = "mirror://luarocks/cassowary-2.3.2-1.rockspec";
+    url    = "https://raw.githubusercontent.com/rocks-moonscript-org/moonrocks-mirror/master/cassowary-2.3.2-1.rockspec";
```
Comment (teto) [inline pkgs/development/lua-modules/generated-packages.nix:187]: damn this updater that keeps chaning the urls, if you wanna fix this, please go ahead, meanwhile can we avoid the change ? with `git add -p` for instance

**Comment by Scoder12** [comment]
It seems the most critical bug is the duplicate argument one

**Comment by Scoder12** [comment]
I removed all the updates that removed mirror URLs so this should be good for now

**Comment by teto** [comment]
could you motivate the update ?  we should be able to bump dependencies at anytime but it gives a bit of context.

could you run `nixpkgs-review pr 185723` on it ?
Seems like the basexx bump broke some things:
```
can you 
       > Error: File not found: dist/basexx-0.4.1-1.rockspec
       For full logs, run 'nix log /nix/store/2a206as9wg4fjlmsv85201n9h3lp0z7f-luajit-2.1.0-2022-04-05-basexx-0.4.1-1.drv'.
error: 1 dependencies of derivation '/nix/store/9h3pa1rs2gj2ayg7m382jmlg00b2zpwq-lua5.3-http-0.3-0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/yhrar6cgshbxrlsp8q3h1ap5bmwp7gck-lua5.2-http-0.3-0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/9d1v5j8y4b5r6895z4m0148lirdjwvpz-lua5.1-http-0.3-0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/f2j7985acjg2aynrhqn9syga45idl2f4-luajit-2.1.0-2022-04-05-http-0.3-0.drv' failed to build
error: 2 dependencies of derivation '/nix/store/0ndpypg3qaypgw56sx0s4b8xiywsv83m-knot-resolver-5.5.1.drv' failed to build
error: 9 dependencies of derivation '/nix/store/8q74grf4v4xd242321gzpjq4lrgq11l0-env.drv' failed to build
error: 1 dependencies of derivation '/nix/store/v9759qzxcj7d36j6d5v5nyzay0p0297g-review-shell.drv' failed to build

Link to currently reviewing PR:
https://github.com/NixOS/nixpkgs/pull/185723

7 packages marked as broken and skipped:
lua51Packages.digestif lua52Packages.digestif lua52Packages.gitsigns-nvim lua53Packages.gitsigns-nvim luaPackages.digestif luaPackages.gitsigns-nvim luajitPackages.digestif

9 packages failed to build:
knot-resolver lua51Packages.basexx lua51Packages.http luaPackages.basexx luaPackages.http lua53Packages.basexx lua53Packages.http luajitPackages.basexx luajitPackages.http

92 packages built:
```

**Comment by Scoder12** [comment]
> could you motivate the update ?

Truthfully it was just because I had to run the update script anyway for my other PRs modifying the generated file. 

> Seems like the basexx bump broke some things

I could just remove that one from the diff. What's the best course of action here?

**Comment by teto** [comment]
> I could just remove that one from the diff. What's the best course of action here?

Depends on your energy level. It's not normal that the version changed from scm to something else
```
 -version = "scm-0";
 +version = "0.4.1-1";
 ```
Apparently I had forked basexx to fix the rockspec url (  https://github.com/teto/basexx/commit/b4ee13f6503499dc52c2d349a81826ee933db43c ) and pointed the update script to my fork in maintainers/scripts/luarocks-packages.csv . This means the update script ignored my custom source, probably a regression somewhere.
 
My change was merged so we should now point at basexx git repo instead of mine. 

Or you could leave it at that since you've done a lot already.

**Comment by teto** [comment]
you can bump luarocks-nix with `nix-shell maintainers/scripts/update.nix --argstr path luarocks-nix` and then regenerate the set

**Comment by Scoder12** [comment]
I've been a bit busy lately but I'm trying my best to get this merged

**Comment by Scoder12** [comment]
Separate PR to bump luarocks-nix?

**Comment by Scoder12** [comment]
I don't understand the error...

**Review by vcunat — CHANGES_REQUESTED** [review]
I believe `fetchTarball` isn't suitable for nixpkgs usage like this.  The reason is that it happens during evaluation.  So even `--dry-run` and similar queries will trigger it, but at that point we don't want them to happen.  That's also why OfBorg fails.

**Comment by Scoder12** [comment]
Ah, makes sense. So what fetcher would work?

**Comment by Scoder12** [comment]
Would fetchzip from nixpkgs work?

**Comment by vcunat** [comment]
Yes, `fetchzip` is OK in this aspect and very close in `fetchTarball` in what it does (hashing unpacked archive).  But I'd recommend its specializations like `fetchFromGitHub` (for cases where they apply).

**Comment by teto** [comment]
> The reason is that it happens during evaluation. So even --dry-run and similar queries will trigger it, but at that point we don't want them to happen.

@vcunat you mean fetchTarball is doing some external computation during eval while fetchZip doesnt ? even if you have a hash all set ? I have trouble finding info about this. @Scoder12 sry I sent you on the wrong track :'( but now you have merge rights on luarocks-nix it should be ok to swap fetchTarball with fetchZip.

**Comment by vcunat** [comment]
It is completely executed during evaluation, i.e. it downloads, unpacks and checks hash (if the hash is specified; it's optional in there).

**Comment by Scoder12** [comment]
@teto was this superseded by #189659 ?

**Comment by teto** [comment]
I am trying to address a few vim/lua issues before the 22.11 release. you can probably rebase the lsqlite3 PR.

**Pull request closed without merging**